src/middleware.js

```diff
@@ -342,12 +336,16 @@ function isValidAction(action) {
     return false;
   }
   // Validate payload for payload-specific action, ignore others
-  if (!actionsWithoutPayload.has(action.type)
-    && !_.has(action, 'payload.data')) {
-    // TODO: move this into standardizer specific area once json standardization
-    // is supported
-    console.error('Payload Data is invalid, expecting payload.data.');
-    return false;
+  if (shouldContainPayload(action)) {
+    // TODO: move this into standardizer specific area once json standardization is supported
```

Comment:
Yes, there is a lot of logic in this file that is related to json-api standard. We should try to generalize this middleware so that json-api becomes equivalent to plain json, firebase, etc.